client/apps/settings/client/add_tag.coffee

```diff
@@ -0,0 +1,22 @@
+React = require 'react'
+ReactDOM = require 'react-dom'
+{ div, input, button } = React.DOM
+
+module.exports = AddTag = React.createClass
+  displayName: 'TagsView'
+
+  addTag: ->
+    @props.addTag @refs.addTagInput.value
+    @refs.addTagInput.value = ''
```

Comment:
I believe using state is the recommended method over refs - this is so concise that it doesn't seem pressing to give a state to this form.
https://facebook.github.io/react/docs/refs-and-the-dom.html#legacy-api-string-refs